allennlp/modules/augmented_lstm.py

```diff
@@ -1,147 +1,228 @@
+            ]
+            timestep_output = (
+                highway_gate * timestep_output
+                + (1 - highway_gate) * highway_input_projection  # noqa
+            )
+
+        return timestep_output, memory
+
+
+class AugmentedLstm(torch.nn.Module):
+    """
+    `AugmentedLstm` implements a one-layer single directional
+    AugmentedLSTM layer. AugmentedLSTM is an LSTM which optionally
+    appends an optional highway network to the output layer. Furthermore the
+    dropout controlls the level of variational dropout done.
+
+    Args:
+        input_size (int): The number of expected features in the input.
+        hidden_size (int): Number of features in the hidden state of the LSTM.
+            Defaults to 32.
+        go_forward (bool): Whether to compute features left to right (forward)
+            or right to left (backward).
+        recurrent_dropout_probability (float): Variational dropout probability
+            to use. Defaults to 0.0.
+        use_highway (bool): If `True` we append a highway network to the
+            outputs of the LSTM.
+        use_input_projection_bias (bool): If `True` we use a bias in
+            our LSTM calculations, otherwise we don't.
+
+    Attributes:
+        cell (AugmentedLSTMCell): AugmentedLSTMCell that is applied at every timestep.
+    """
+
+    def __init__(
+        self,
+        input_size: int,
+        hidden_size: int,
+        go_forward: bool = True,
+        recurrent_dropout_probability: float = 0.0,
+        use_highway: bool = True,
+        use_input_projection_bias: bool = True,
+    ):
+        super().__init__()
+
+        self.embed_dim = input_size
+        self.lstm_dim = hidden_size
+
+        self.go_forward = go_forward
+        self.use_highway = use_highway
+        self.recurrent_dropout_probability = recurrent_dropout_probability
+
+        self.cell = AugmentedLSTMCell(
+            self.embed_dim, self.lstm_dim, self.use_highway, use_input_projection_bias
+        )
+
+    def forward(
+        self, inputs: PackedSequence, states: Optional[Tuple[torch.Tensor, torch.Tensor]] = None
+    ) -> Tuple[PackedSequence, Tuple[torch.Tensor, torch.Tensor]]:
+        """
+        Warning: Would be better to use the BiAugmentedLstm class in a regular model
+
+        Given an input batch of sequential data such as word embeddings, produces
+        a single layer unidirectional AugmentedLSTM representation of the sequential
+        input and new state tensors.
+
+        Args:
+            inputs (PackedSequence): Input tensor of shape
+                (bsize x seq_len x input_dim).
+            states (Tuple[torch.Tensor, torch.Tensor]): Tuple of tensors containing
+                the initial hidden state and the cell state of each element in
+                the batch. Each of these tensors have a dimension of
+                (1 x bsize x num_directions * nhid). Defaults to `None`.
```

Comment:
There is only one direction in `AugmentedLstm`, isn't there? It looks like `num_directions` belongs in the class below?